source/features/clean-conversation-headers.tsx

```diff
@@ -26,25 +30,31 @@ function initIssue(): void {
 
 function initPR(): void {
 	const observer = observe('.gh-header-meta .flex-auto:not(.rgh-clean-conversation-header)', {
-		add(byline) {
+		async add(byline) {
 			byline.classList.add('rgh-clean-conversation-header');
-			const isSameAuthor = select('.js-discussion > .TimelineItem:first-child .author')?.textContent === select('.author', byline)!.textContent;
-			const baseBranch = select('.commit-ref:not(.head-ref)', byline)!;
-			const isDefaultBranch = (baseBranch.firstElementChild as HTMLAnchorElement).pathname.split('/').length === 3;
 
-			// Removes: [octocat wants to merge 1] commit into github:master from octocat:feature
-			// Removes: [octocat] merged 1 commit into master from feature
-			for (const node of [...byline.childNodes].slice(isSameAuthor ? 0 : 2, pageDetect.isMergedPR() ? 2 : 4)) {
+			const author = select('.author', byline)!;
+			const isSameAuthor = pageDetect.isPRConversation() && author.textContent === select('.TimelineItem .author')!.textContent;
+
+			const base = select('.commit-ref', byline)!;
+			const baseBranch = base.title.split(':')[1];
+			const isDefaultBranch = baseBranch === await getDefaultBranch() || (pageDetect.isClosedPR() && baseBranch === 'master');
+
+			byline.childNodes[pageDetect.isClosedPR() ? (pageDetect.isMergedPR() ? 5 : 7) : 9].replaceWith(<> <ArrowLeftIcon/> </>);
```

Comment:
Oof that’s quite a line. Let’s `select` where possible instead of using indexes if the intention is readability.

Reply:
I must have confused myself 😅 this should be replaced with a select(). previousSibling